### ISSUE TITLE
Remove false positive from ExpectedResult when used in connection with async tests

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit.analyzers#50]
The following tests are correct, but the analyzer gives the following warning: **The ExpectedResult value cannot be assigned to the return type, Task`1**.

```csharp
    [TestCase(ExpectedResult = 1)]
    public async Task<int> AsyncGenericTaskTestCaseWithExpectedResult()
    {
      return await Task.Run(() => 1);
    }
```
and 
```csharp
    [TestCase(ExpectedResult = 1)]
    public Task<int> GenericTaskTestCaseWithExpectedResult()
    {
      return Task.Run(() => 1);
    }
```
